base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1250,20 +1233,17 @@ where
                 let self_signature = self
                     .resources
                     .key_manager
-                    .sign_script_message(&self.resources.wallet_identity.wallet_node_key_id, &script_challange)
+                    .sign_script_message(&self.resources.key_manager.get_spend_key().await?.0, &script_challange)
                     .await?;
-                script_input_shares.insert(
-                    self.resources.wallet_identity.address.public_spend_key().clone(),
-                    self_signature,
-                );
+                script_input_shares.insert(self.resources.key_manager.get_spend_key().await?.1, self_signature);
 
                 // the order here is important, we need to add the signatures in the same order as public keys where
                 // added to the script originally
                 for key in public_keys {
                     if let Some(signature) = script_input_shares.get(&key) {
                         script_signatures.push(StackItem::Signature(signature.clone()));
                         // our own key should not be added yet, it will be added with the script signing
-                        if &key != self.resources.wallet_identity.address.public_spend_key() {
+                        if key != self.resources.key_manager.get_spend_key().await?.1 {
```

Comment:
Should we have a custom type for returning a `(P, PK)` instead of using tuples and indexes?

Reply:
This is not an idea I want to dismiss out of hand. I remember us looking into it the first time we did this pattern for the key manager, and then decided that for two returns only its perhaps worth it just to return a tuple.